src/app/_icons/icons.tsx

```diff
@@ -1,25 +1,45 @@
-export function MagnifyingGlass({ x, y, width = '24', height = '24', stroke = 'black' }) {
+interface IconProperties {
+  x?: string
+  y?: number
+  width?: string
+  height?: string
+  color?: string
+}
+
+export function MagnifyingGlass({
+  x,
+  y,
+  width = '24',
+  height = '24',
+  color = 'var(--dark-rock-800)',
+}: IconProperties) {
   return (
     <svg
       x={x}
       y={y}
-      width="24"
-      height="24"
+      width={width}
+      height={height}
       viewBox="0 0 24 24"
       fill="none"
       xmlns="http://www.w3.org/2000/svg"
     >
       <path
         d="M16.927 17.0401L20.4001 20.4001M19.2801 11.4401C19.2801 15.77 15.77 19.2801 11.4401 19.2801C7.11019 19.2801 3.6001 15.77 3.6001 11.4401C3.6001 7.11019 7.11019 3.6001 11.4401 3.6001C15.77 3.6001 19.2801 7.11019 19.2801 11.4401Z"
-        stroke="black"
+        stroke={color}
```

Comment:
A useful feature of svg is that you can define `fill="currentColor"`, and it will inherit the text color from its ancestors which is a nice default. This works though 👍

Reply:
oh! so this means the color parameter is not necessary on conditionally changing fields?
if i use the svg on a <p> and use currentColor it'll "absorb" the `color` attribute? nice